react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      image: 1
+    };
+    this.handleIndicatorClick = this.handleIndicatorClick.bind(this);
+    this.handleArrowClick = this.handleArrowClick.bind(this);
+    this.rotate = this.rotate.bind(this);
+    this.interval = setInterval(this.rotate, 3000);
```

Comment:
We generally start timers, network requests, and DOM manipulation in `componentDidMount()`